src/Algebra/Graph/NonEmpty.hs

```diff
@@ -377,9 +377,27 @@ hasVertex v = foldg1 (==v) (||) (||)
 -- hasEdge x y . 'removeEdge' x y == const False
 -- hasEdge x y                  == 'elem' (x,y) . 'edgeList'
 -- @
-hasEdge :: Ord a => a -> a -> NonEmptyGraph a -> Bool
+hasEdge :: Eq a => a -> a -> NonEmptyGraph a -> Bool
 hasEdge = T.hasEdge
 
+-- | Check if a graph contains a given loop.
+-- Complexity: /O(s)/ time.
+--
+-- @
+-- hasLoop x 'empty'            == False
+-- hasLoop x ('vertex' z)       == False
```

Comment:
Why `z` instead of `y`? Seems surprising :)

Reply:
Again, copy/paste and some suppression of `y` from `hasEdge` ^^ corrected